### PR TITLE
Adding service.environment to the spec

### PR DESCRIPTION
**Pull request by eyalkoren**
There is no `service.environment` field in ECS at the moment, but there is an [RFC](https://github.com/elastic/ecs/blob/master/rfcs/text/0002-rfc-environment.md) to add it.
This PR is blocked until it is added.

### CODE REVIEW

**Comment by ghost** [comment]
## :green_heart: Build Succeeded



> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[Pipeline View](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fecs-logging-update-specs-mbp%2FPR-62/detail/PR-62/96//pipeline) [Test View](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fecs-logging-update-specs-mbp%2FPR-62/detail/PR-62/96//tests) [Changes](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fecs-logging-update-specs-mbp%2FPR-62/detail/PR-62/96//changes) [Artifacts](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fecs-logging-update-specs-mbp%2FPR-62/detail/PR-62/96//artifacts) [preview](http://ecs-logging_62.docs-preview.app.elstc.co/diff)  [preview](https://ci-stats.elastic.co/app/apm/services/apm-ci/transactions/view?rangeFrom=2022-05-16T05:38:00.380Z&rangeTo=2022-05-16T05:58:00.380Z&transactionName=BUILD+apm-shared%2Fecs-logging-update-specs-mbp%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=4afab055833fc8a6a75f24ac2e42bdd7&transactionId=a458d99487aedb5a)


<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2022-05-16T05:48:00.380+0000


* Duration: 7 min 32 sec





</p>
</details>

**Comment by felixbarny** [comment]
@cyrille-leclerc is there a timeline for adding `service.environment` to ECS?

File: spec/spec.json [hunk 1]
```diff
@@ -93,6 +93,14 @@
                 "The field helps to filter for different log streams from the same pod, for example and is required for log anomaly detection."
             ]
         },
+        "service.environment": {
+            "type": "string",
+            "required": false,
+            "url": "https://www.elastic.co/guide/en/ecs/current/ecs-service.html",
+            "comment": [
+                "Configurable by users."
```
Comment (felixbarny) [inline spec/spec.json]:
```suggestion
                "Configurable by users.",
                "When an APM agent is active, it should auto-configure it if not already set."
```

File: spec/spec.json [hunk 2]
```diff
@@ -93,6 +93,14 @@
                 "The field helps to filter for different log streams from the same pod, for example and is required for log anomaly detection."
             ]
         },
+        "service.environment": {
+            "type": "string",
+            "required": false,
+            "url": "https://www.elastic.co/guide/en/ecs/current/ecs-service.html",
```
Comment (felixbarny) [inline spec/spec.json]:
```suggestion
            "url": "https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-environment",
```

**Review by felixbarny — COMMENTED** [review]
The service.environment field is in ECS now, let's move this out of draft

**Comment by eyalkoren** [comment]
@Mergifyio update

**Review by felixbarny — APPROVED** [review]

**Pull request merged**